app/backend/src/tests/test_discussions.py

```diff
@@ -0,0 +1,169 @@
+        session.add(node)
+        community_cluster = Cluster(
+            name=f"Testing Community",
+            description=f"Description for testing community",
+            parent_node=node,
+            official_cluster_for_node=node,
+            thread=Thread(),
+        )
+        session.add(community_cluster)
+        main_page = Page(
+            creator_user_id=user.id,
+            owner_cluster=community_cluster,
+            type=PageType.main_page,
+            main_page_for_cluster=community_cluster,
+            thread=Thread(),
+        )
+        session.add(main_page)
+        session.add(
+            PageVersion(
+                page=main_page,
+                editor_user_id=user.id,
+                title=f"Main page for the testing community",
+                content="Empty.",
+            )
+        )
+        # create a group
+        group_cluster = Cluster(
+            name=f"Testing Group",
+            description=f"Description for testing group",
+            parent_node=node,
+            thread=Thread(),
+        )
+        session.add(group_cluster)
+        main_page = Page(
+            creator_user_id=user.id,
+            owner_cluster=group_cluster,
+            type=PageType.main_page,
+            main_page_for_cluster=group_cluster,
+            thread=Thread(),
+        )
+        session.add(main_page)
+        session.add(
+            PageVersion(
+                page=main_page,
+                editor_user_id=user.id,
+                title=f"Main page for the testing community",
+                content="Empty.",
+            )
+        )
+        session.flush()
+
+        community_id = node.id
+        community_cluster_id = community_cluster.id
+        group_id = group_cluster.id
+
+    with discussions_session(token) as api:
+        res = api.CreateDiscussion(
+            discussions_pb2.CreateDiscussionReq(
+                title="dummy title",
+                content="dummy content",
+                owner_community_id=community_id,
+            )
+        )
+
+        assert res.title == "dummy title"
+        assert res.content == "dummy content"
+        assert res.slug == "dummy-title"
+        assert to_aware_datetime(res.created) > now() - timedelta(seconds=10) and to_aware_datetime(res.created) < now()
+        assert res.creator_user_id == user.id
+        assert res.owner_community_id == community_id
+
+        discussion_id = res.discussion_id
+
+    with discussions_session(token) as api:
+        res = api.GetDiscussion(
+            discussions_pb2.GetDiscussionReq(
+                discussion_id=discussion_id,
+            )
+        )
+
+        assert res.title == "dummy title"
+        assert res.content == "dummy content"
+        assert res.slug == "dummy-title"
+        assert to_aware_datetime(res.created) > now() - timedelta(seconds=10) and to_aware_datetime(res.created) < now()
```

Comment:
Use Python's chained comparisons and have a more generous time limit, we don't want random failures because of loaded servers. You could set `test_start_time = now()` before creating the discussion and then change this to
```
assert test_start_time <= to_aware_datetime(res.created) <= now()
```

Reply:
Thanks, great idea. I've cleaned it up